Esri/arcgis-maps-sdk-swift-samples
# [New] List spatial reference transformations

## Description

This PR implements `List spatial reference transformations` in `Edit and Manage Data` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-ListSpatialReferenceTransformations/Shared/Samples/List%20spatial%20reference%20transformations)

## Linked Issue(s)

- `swift/issues/4921`

## How To Test

- Tap on a transformation to see the projected geometry update.
- Tap "Suitable for Map Extent" limit the transformations to those that are appropriate for the current extent.
- To add projection engine data:
    - Tap "Download Data" and the download button next to the latest release of the `Projection Engine Data`.
    - Unzip the downloaded data in Files.
    - Tap "Set Data Directory" in the sample.
    - Navigate **INTO** the unzipped `Projection Engine Data` directory and tap "Open".

**NOTE:** Adding projection engine data while on-simulator causes it to crash. This currently only works on-device.

## Screenshots

![list-spatial-reference-transformations](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/9f067f6a-6bde-47d1-a6ed-d777f9409756)
